ASCII control characters escaping in error messages

Line feed literals, carriage returns, and tabs are all printed as their actual characters rather than the escaped sequence in error messages.

Note in the reproducer that a new line seperates '1' and '2', a sequence of spaces seperates '3' and '4', and that '6' has been printed at the start of the line as the terminal's cursor has interpreted the carriage return to overprint text.

Reproducer:

```ps1con
(sphinx) PS S:\Development\sphinx> ruff -V
ruff 0.0.199
(sphinx) PS S:\Development\sphinx> type bug.py
if token == "1\n2":
    pass

if token == "3\t4":
    pass

if token == "5\r6":
    pass
(sphinx) PS S:\Development\sphinx> ruff bug.py --select S105
bug.py:1:13: S105 Possible hardcoded password: `"1
2"`
bug.py:4:13: S105 Possible hardcoded password: `"3      4"`
6"`.py:7:13: S105 Possible hardcoded password: `"5
Found 3 error(s).
```

A

--- a/resources/test/fixtures/flake8_bandit/S105.py
+++ b/resources/test/fixtures/flake8_bandit/S105.py
@@ -51,3 +51,12 @@ secret == "s3cr3t"
 token == "s3cr3t"
 secrete == "s3cr3t"
 password == safe == "s3cr3t"
+
+if token == "1\n2":
+    pass
+
+if token == "3\t4":
+    pass
+
+if token == "5\r6":
+    pass

--- a/src/checks.rs
+++ b/src/checks.rs
@@ -2850,14 +2850,13 @@ impl CheckKind {
             CheckKind::HardcodedBindAllInterfaces => {
                 "Possible binding to all interfaces".to_string()
             }
-            CheckKind::HardcodedPasswordString(string) => {
-                format!("Possible hardcoded password: `\"{string}\"`")
-            }
-            CheckKind::HardcodedPasswordFuncArg(string) => {
-                format!("Possible hardcoded password: `\"{string}\"`")
-            }
-            CheckKind::HardcodedPasswordDefault(string) => {
-                format!("Possible hardcoded password: `\"{string}\"`")
+            CheckKind::HardcodedPasswordString(string)
+            | CheckKind::HardcodedPasswordFuncArg(string)
+            | CheckKind::HardcodedPasswordDefault(string) => {
+                format!(
+                    "Possible hardcoded password: `\"{}\"`",
+                    string.escape_debug()
+                )
             }
             // flake8-blind-except
             CheckKind::BlindExcept(name) => format!("Do not catch blind exception: `{name}`"),

